metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -973,11 +1178,44 @@ def get_schema_metadata_for_custom_sql(
         )
         return schema_metadata
 
-    def _get_project(self, node):
-        if node.get("__typename") == "EmbeddedDatasource" and node.get("workbook"):
-            return node["workbook"].get("projectName")
-        elif node.get("__typename") == "PublishedDatasource":
-            return node.get("projectName")
+    def _get_project(self, ds):
+        # For PublishedDatasource the GraphQL API doesn't return the projectLuid and hence we need to fetch
+        # the projectLuid from datasource_project_map
+        # For EmbeddedDatasource the GraphQL API returns the projectLuid in resposne and
+        # hence it is available at ds["workbook"]["projectLuid"]
+        if self.config.extract_project_hierarchy:
+            if (
+                ds.get("__typename") == "PublishedDatasource"
+                and ds.get("luid")
+                and ds["luid"] in self.datasource_project_map.keys()
+                and self.datasource_project_map[ds["luid"]]
+                in self.tableau_project_registry.keys()
+            ):
+                return self._get_project_browse_path(
+                    self.datasource_project_map[ds["luid"]]
+                )
+            elif (
```

Comment:
"""
This method extracts the project LUID based on whether it's an Embedded Data Source or a Published Data Source.

If it's a published data source, this comes from x.
If it's an embedded data source, this comes from y. 
"""
extract_project_luid_from_datasource: 
     .....